### DISCUSSION TITLE
セキュリティについて

### DISCUSSION THREAD

**Discussion by sugawa0718** [FisaCloudAlliance/JamstackEducation#60] [Q&A]
GitHubで開発するにあたり、ソースコードの流出などセキュリティが心配なのですが、何か対策等されていますか？

**Comment by s-ueno** [comment]
ご質問、ありがとうございます！

いくつかポイントがありますので、ご紹介できればと思います。

### GitHubには、Ownerという管理者権限で様々な設定が可能です。（無料でも可能！）

たとえば、利用者のログイン時に2FA（二要素認証）を強制するポリシーや、ソースコードは必ず非公開でしか作成できないなど、様々な設定が可能です。

今回はオープンな勉強サイトのため、全てのポリシーをオフで利用していますが、企業向けの場合は、これらのポリシーを管理者が設定することで、ソースコードを安全に運用することが可能になります。


### GitHub Enterprise の利用

弊社では、GitHub Enterprise を利用しております。上記に加えて、社内のActiveDirectoryとアカウントを連動させることなども可能で、より安全で柔軟な設定が可能となります。


### AIによる自動セキュリティチェック機能（無料でも可）

某企業さまで情報漏洩の件が最近にぎわせていますが、個人情報など乗せてはならない情報をGitHubに謝ってチェックインしたとしても、その瞬間に自動で検知して管理者へ即時アラートを投げる機能がGitHubにはデフォルトで付随しております。

これらの機能を利用することで、不意の事故にも即座に対応可能です。


最後になりましたが、GitHub社の最新の情報について記事を共有いたします。
　※弊社も日本のGitHub Enterpriseユーザー企業の例として[イベントで紹介](https://cloud.watch.impress.co.jp/img/clw/docs/1370/876/html/github05_o.jpg.html)させていただきました！

https://cloud.watch.impress.co.jp/docs/news/1370876.html

**Reply by sugawa0718** [reply]
@s-ueno 
早々に回答いただきありがとうございます！

GitHub、企業も無料で使える枠があるのですね。

>某企業さまで情報漏洩の件が最近にぎわせていますが、個人情報など乗せてはならない情報をGitHubに謝ってチェックインしたとしても、その瞬間に自動で検知して管理者へ即時アラートを投げる機能がGitHubにはデフォルトで付随しております。

まさにその話題がでていたので、どのように対策したらいいのかと思っていました。

GitHubにはセキュリティを担保できる機能があるので、いかに運用するかが大切だと理解しました。

弊社でも利用できないか検討してみます。
ありがとうございました。
